spec/lib/lighthouse/auth/client_credentials/jwt_generator_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'lighthouse/auth/client_credentials/jwt_generator'
+
+RSpec.describe Auth::ClientCredentials::JWTGenerator do
+  let(:public_key) { OpenSSL::PKey::RSA.new(File.read('spec/support/certificates/lhdd-public.pem')) }
+
+  before do
+    allow(Auth::ClientCredentials::JWTGenerator).to receive(:build_rsa_instance).and_return(
+      OpenSSL::PKey::RSA.new(
+        File.read('spec/support/certificates/lhdd-private.pem')
+      )
+    )
+  end
+
+  it 'generates a valid JWT token for direct deposit management API' do
+    client_id = Settings.lighthouse.direct_deposit.access_token.client_id
```

Comment:
I honestly would just put values directly in here that way you don't couple the tests for this module too closely to the direct deposit module. This would allow you to put `~` for `client_id` and `rsa_key` instead of having to put in fake values just to make these tests work